### PR TITLE
Put 1.25.3 docs in the repo

### PR DESCRIPTION
**Pull request by blairconrad**
Connects to #626.

Docs can be previewed at http://bfakeiteasy.readthedocs.org/en/read-the-docs-1.25.3/

Imported the documentation from the wiki, in a MkDocs-style layout.
Adjusted the markdown to conform to MkDocs/ReadTheDocs requirements,
removed references to features in 2.0, and
corrected some minor errors.

### CODE REVIEW

**Comment by blairconrad** [comment]
Oh, ordinarily I'd've preferred to have a series of commits in this PR, but I was undisciplined in my conversion/fixup process, and didn't think I could separate out commits without essentially doing the whole thing again.
If you prefer, I can do the whole thing again.

File: docs/scanning-for-extension-points.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+On initialization, essentially as soon as a FakeItEasy type is
+accessed, FakeItEasy uses reflection to look for internal and
+user-supplied extension points. In most cases, there is no _need_ for
+users to define any extensions, but they may be used to enhance the
+power and usability of FakeItEasy.
+
+There are currently three kinds of extension points defined:
+
+* [Custom Dummy Creation](custom-dummy-creation.md) rules,
+* Fake Configurators
```
Comment (thomaslevesque) [inline docs/scanning-for-extension-points.md]: Wasn't there a wiki page about fake configurators? Can't find it now, but I seem to remember it was there at some point...

**Comment by thomaslevesque** [comment]
Looks good to me. Of course, I didn't read everything in detail, since most of it is existing content from the wiki. I just checked that there was no reference to 2.0 features, and saw no obvious mistake.

File: docs/scanning-for-extension-points.md (hunk 1 above)
Comment (blairconrad) [inline docs/scanning-for-extension-points.md]: Actually, there never was! Hard to believe, I know, but we didn't have one.
In fact, I had only the most tenuous of ideas of what a "Fake Configurator" was until, well, pretty near the time we started on 2.0 work, I think.
I'm lazy and figured that everyone had gotten along okay without such a page up 'til now, so I didn't bother making one for the conversion.
If you'd like, I can write something up.

**Comment by blairconrad** [comment]
Thanks, yeah. I can't even say that I read _everything in detail_ either. I read it, but it's pretty easy to gloss over things. There may be mistakes, but at least I don't think there are any places that confuse 1.x functionality with 2.x. And probably most of the mistakes were already there…

We can always keep this open a bit longer in case @adamralph wants to take a look. If nothing else, you've given me the confidence I need to proceed with the 2.0 PR!

File: docs/scanning-for-extension-points.md (hunk 1 above)
Comment (thomaslevesque) [inline docs/scanning-for-extension-points.md]: > If you'd like, I can write something up.

I don't think it's worth the trouble. Most users will probably want to use the 2.0 version anyway.

Comment (blairconrad) [inline docs/scanning-for-extension-points.md]: And even if not, I think _we_ want most uses to use the 2.0 version! :wink:

**Comment by adamralph** [comment]
LGTM, but I don't think we should be merging this into a branch named `1.25.3-docs`. Effectively, we are performing a hack 1.x and if, we ever wanted to do an actual patch 1.25.3, we'd want to do this on the same branch, so I think we should call this branch `support-1` (borrowing from gitlfow).

**Comment by blairconrad** [comment]
I'm not really fussy about the naming, and we can certainly rename, but could you point me to a reference to the `support-1` naming scheme? I couldn't find it at [A successful Git Branching Model](http://nvie.com/posts/a-successful-git-branching-model/).

Also, I'm not entirely sure why we need to keep the branch around after we move the 1.25.3 tag to the commit that has the merge, so am not sure the name matters so much…

**Comment by adamralph** [comment]
The `support-` branch concept has never made it past "experimental" in gitflow. We use it at Particular for NServiceBus to support patching, and back porting features to, older versions. There is some info on it around the interwebs, e.g. https://groups.google.com/forum/#!topic/gitflow-users/I9sErOSzYzE

We have to keep the branch around. If we delete it, then we delete the commits that only it contains, including the one that is tagged.

**Comment by blairconrad** [comment]
> We have to keep the branch around. If we delete it, then we delete the commits that only it contains, including the one that is tagged.

Oh? That doesn't jive with the behaviour I've seen.
Upstream has only these branches:

* 1.25.3-docs
* AssertionExceptions
* NSubstituteSpike
* master

And I created 1.25.3-docs from the 1.25.3 tag last week…

**Comment by adamralph** [comment]
All the current tags live on one or more branches. The 1.25.3 tag lives on the master branch and branches which originate from it. If you were to delete those branches, you'd lose all the commits and hence the tags.

Note that tags in got are very different to tags in, say, SVN. In SVN there is effectively no difference, but in git, a tag belongs to a commit, and it's existence depends on the existence of that commit.

**Comment by blairconrad** [comment]
Aha! When the 1.25.3 tag was created, it wasn't "on master". It was on some other crazy hotfix branch whose name I've fogotten.
Now, 1.25.3 is "on master" because we merged the commit that was tagged with 1.25.3 into master.
I get it!

**Comment by blairconrad** [comment]
Closing in favour of #634, which attempts to merge into the (new) support-1 branch.

**Pull request closed without merging**